pkg/pipeline/PipelineBuilder.go

```diff
@@ -1880,7 +1901,7 @@ func (impl PipelineBuilderImpl) PatchCdPipelines(cdPipelines *bean.CDPatchReques
 	}
 }
 
-func (impl PipelineBuilderImpl) DeleteCdPipeline(pipeline *pipelineConfig.Pipeline, ctx context.Context, deleteAction int, deleteFromAcd bool, userId int32) (*bean.AppDeleteResponseDTO, error) {
+func (impl PipelineBuilderImpl) DeleteCdPipeline(pipeline *pipelineConfig.Pipeline, ctx context.Context, deleteAction int, deleteFromAcd bool, userId int32, cdPipelinePluginDeleteReq *bean.CDPipelineConfigObject) (*bean.AppDeleteResponseDTO, error) {
```

Comment:
why do we need to pass ``cdPipelinePluginDeleteReq`` as argument, why can't this func extract this value within itself ??

Reply:
yes refactored it